File: app/src/main/java/com/apollopharmacy/vishwam/ui/home/qcfail/dashboard/DashBoardFragment.java
package com.apollopharmacy.vishwam.ui.home.qcfail.dashboard;


import androidx.annotation.NonNull;

import androidx.recyclerview.widget.LinearLayoutManager;

import android.util.Log;

import com.anychart.AnyChart;

import com.anychart.chart.common.dataentry.DataEntry;
import com.anychart.chart.common.dataentry.ValueDataEntry;
import com.anychart.chart.common.listener.Event;
import com.anychart.chart.common.listener.ListenersInterface;
import com.anychart.charts.Pie;
import com.apollopharmacy.vishw.PendingFragment;
import com.apollopharmacy.vishwam.R;
import com.apollopharmacy.vishwam.base.BaseFragment;
import com.apollopharmacy.vishwam.databinding.FragmentDashBoardBinding;
import com.apollopharmacy.vishwam.ui.home.qcfail.approved.ApprovedFragment;
import com.apollopharmacy.vishwam.ui.home.qcfail.dashboard.adapter.SiteTableAdapter;
import com.apollopharmacy.vishwam.ui.home.qcfail.dashboard.model.SiteData;
import com.apollopharmacy.vishwam.ui.home.qcfail.rejected.RejectedFragment;

import java.util.ArrayList;
import java.util.List;

public class DashBoardFragment extends BaseFragment<QcDashBoardViewModel, FragmentDashBoardBinding> {
    ArrayList<SiteData> siteDataList = new ArrayList<>();

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_dash_board;
    }

    @NonNull
    @Override
    public QcDashBoardViewModel retrieveViewModel() {
        return new QcDashBoardViewModel();
    }

    @Override
    public void setup() {
        siteDataList.add(new SiteData("16001", "24", "12", "7", "5"));
        siteDataList.add(new SiteData("14055", "71", "16", "49", "6"));
        siteDataList.add(new SiteData("14068", "580", "266", "211", "103"));
        siteDataList.add(new SiteData("12357", "45", "12", "20", "13"));

        SiteTableAdapter siteTableAdapter = new SiteTableAdapter(siteDataList, getContext());
        viewBinding.siteTableRcv.setLayoutManager(new LinearLayoutManager(getContext()));
        viewBinding.siteTableRcv.setAdapter(siteTableAdapter);

        // Pie chart
        Pie pie = AnyChart.pie();
        pie.labels().format("{%value}");

        List<DataEntry> data = new ArrayList<>();
        data.add(new ValueDataEntry("Pending", 257));
        data.add(new ValueDataEntry("Approved", 321));
        data.add(new ValueDataEntry("Rejected", 142));
        pie.data(data);

        viewBinding.pieChart.setChart(pie);


        pie.setOnClickListener(new ListenersInterface.OnClickListener(new String[]{"x", "value"}) {
            @Override
            public void onClick(Event event) {
                Log.i("PIE_CHART", "onClick: " + event.getData().get("value"));
                Log.i("PIE_CHART", "onClick: " + event.getData().get("x"));
                if (event.getData().get("x").equals("Pending")) {
                    getFragmentManager().beginTransaction().replace(R.id.fragment_container, new PendingFragment()).commit();
                } else if (event.getData().get("x").equals("Approved")) {
                    getFragmentManager().beginTransaction().replace(R.id.fragment_container, new ApprovedFragment()).commit();
                } else if (event.getData().get("x").equals("Rejected")) {
                    getFragmentManager().beginTransaction().replace(R.id.fragment_container, new RejectedFragment()).commit();
                }
            }
        });
    }
}

File: app/src/main/java/com/apollopharmacy/vishwam/ui/home/qcfail/dashboard/DashboardFragment.kt
package com.apollopharmacy.vishwam.ui.home.qcfail.dashboard

import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.anychart.AnyChart
import com.anychart.AnyChartView
import com.anychart.chart.common.dataentry.DataEntry
import com.anychart.chart.common.dataentry.ValueDataEntry
import com.anychart.chart.common.listener.Event
import com.anychart.chart.common.listener.ListenersInterface
import com.anychart.charts.Pie
import com.anychart.graphics.vector.SolidFill
import com.apollopharmacy.vishw.PendingFragment
import com.apollopharmacy.vishwam.R
import com.apollopharmacy.vishwam.base.BaseFragment
import com.apollopharmacy.vishwam.data.network.LoginRepo
import com.apollopharmacy.vishwam.databinding.FragmentDashboardBinding
import com.apollopharmacy.vishwam.ui.home.qcfail.approved.ApprovedFragment
import com.apollopharmacy.vishwam.ui.home.qcfail.rejected.RejectedFragment
import lecho.lib.hellocharts.model.SliceValue
import java.util.*


class DashboardFragment : BaseFragment<DashBoardViewModel, FragmentDashboardBinding>() {
    private var chart: AnyChartView? = null
    private val count = listOf(257, 321, 142)
    private val lists = listOf("Pending", "Approved", "Reject")
    var pieData: List<SliceValue> = ArrayList()

    override val layoutRes: Int
        get() = R.layout.fragment_dashboard

    override fun retrieveViewModel(): DashBoardViewModel {
        return ViewModelProvider(this).get(DashBoardViewModel::class.java)
    }

    override fun setup() {

        val userData = LoginRepo.getProfile()
        if (userData != null) {
            viewBinding.empname.setText(userData.EMPNAME)
            viewBinding.empid.setText(userData.EMPID)
        }

//

        configChartView()
        var names = ArrayList<String>()

        names.add("5")
        names.add("8")
        names.add("12")
//        viewBinding.dashboardrecycleview.adapter = context?.let { DashboardAdapter(it, names) }

    }

    private fun configChartView() {


        val pie: Pie = AnyChart.pie()
        var names = ArrayList<String>()
        pie.palette().itemAt(2, SolidFill("#c7c7c7", 1
        ))
        pie.palette().itemAt(1, SolidFill("#FF6D6A", 1
        ))
        val dataPieChart: MutableList<DataEntry> = mutableListOf()
        for (index in count.indices) {
            dataPieChart.add(ValueDataEntry(lists.elementAt(index), count.elementAt(index)))
        }
        pie.labels().format("{%value}")

        pie.animation(true, 800)
        pie.data(dataPieChart)

        pie.setOnClickListener(object :
            ListenersInterface.OnClickListener(arrayOf("x", "value")) {
            override fun onClick(event: Event) {
                event.getData().get("x")
                event.getData().get("value")

                if (event.getData().get("x").equals("Approved")) {
                    //APL49396


//                    findNavController().navigate(R.id.action_dashboardFragment_to_approvedFragment)
//
                    val fragment: Fragment = ApprovedFragment()
                    val fragmentManager = activity!!.supportFragmentManager
                    val fragmentTransaction = fragmentManager.beginTransaction()
//                    fragmentTransaction.replace(R.id.fragment_container, fragment)
                    fragmentTransaction.addToBackStack(null)
//                    fragmentTransaction.setReorderingAllowed(true)
//                    fragmentTransaction.isAddToBackStackAllowed
                    fragmentTransaction.commit()
//                    MainActivity.mInstance.close()


                } else if (event.getData().get("x").equals("Pending")) {
                    val fragment: Fragment = PendingFragment()
                    val fragmentManager = activity!!.supportFragmentManager
                    val fragmentTransaction = fragmentManager.beginTransaction()
//                    fragmentTransaction.replace(R.id.fragment_container, fragment)
                    fragmentTransaction.addToBackStack(null)
                    fragmentTransaction.commit()

                } else if (event.getData().get("x").equals("Reject")) {
                    val fragmentManager = activity!!.supportFragmentManager
                    val fragmentTransaction = fragmentManager.beginTransaction()
//                    fragmentTransaction.replace(R.id.fragment_container, RejectedFragment())
                    fragmentTransaction.addToBackStack(null)
                    fragmentTransaction.commit()
                }
            }
        })
        viewBinding.piechart.setChart(pie)

    }


}